packages/store/src/internal/state-factory.ts

```diff
@@ -214,7 +199,7 @@ export class StateFactory {
   /**
    * Create the state context
    */
-  private createStateContext(metadata: MappedStore) {
+  createStateContext(metadata: MappedStore) {
```

Comment:
This should be kept private and `StateContextFactory` should be injected and used where this was being used.
( added comment to usage in lifecycle-state-manager.ts )

Reply:
Also, after moving the `invokeInit` method out of this class the only usage of this method is in the `invokeActions` method. We could actually remove this private method here and call `this._stateContextFactory.createStateContext` directly in the `invokeActions` method.